src/core/Table/TablePaginator.tsx

```diff
@@ -0,0 +1,335 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import '@itwin/itwinui-css/css/table.css';
+import { IconButton } from '../Buttons/IconButton';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import { ButtonGroup } from '../ButtonGroup';
+import { Button } from '../Buttons/Button';
+import { useOverflow } from '../utils/hooks/useOverflow';
+import { ProgressRadial } from '../ProgressIndicators';
+import { TableProps } from './Table';
+import { DropdownButton } from '../Buttons';
+import { MenuItem } from '../Menu';
+import { CommonProps, getBoundedValue, useTheme } from '../utils';
+
+const defaultLocalization = {
+  rowsPerPageLabel: (size: number) => `${size} per page`,
+  currentRowsInfoLabel: (
+    startIndex: number,
+    endIndex: number,
+    totalRows: number,
+    isLoading: boolean,
+  ) =>
+    isLoading
+      ? `${startIndex}-${endIndex}…`
+      : `${startIndex}-${endIndex} of ${totalRows}`,
+};
+
+export type TablePaginatorProps = {
+  /**
+   * The zero-based index of the current page.
+   */
+  currentPage: number;
+  /**
+   * Total number of rows.
+   */
+  totalRowsCount: number;
+  /**
+   * Number of rows per page.
+   */
+  pageSize: number;
+  /**
+   * Callback when page is changed.
+   */
+  onPageChange: (page: number) => void;
+  /**
+   * Callback when page size is changed.
+   */
+  onPageSizeChange?: (size: number) => void;
+  /**
+   * Modify the density of the pagination (adjusts the height).
+   * @default 'default'
+   */
+  density?: TableProps['density'];
```

Comment:
I don't know is it good to accepts `density` when only there are two sizes.

Reply:
What do you mean "only two sizes"?

There are 3 paginator heights here:
https://itwin.github.io/iTwinUI/backstop/minified/table.html#demo-extras

Reply:
It comes from table:
![image](https://user-images.githubusercontent.com/36186912/136050030-dd7fc2c4-7821-4c63-afcc-f58d3d0180cf.png)

In paginator I only apply `small` to buttons when it is `condensed` or `extra-condensed`.